central/graphql/resolvers/loaders/node_cves.go

```diff
@@ -0,0 +1,150 @@
+		return NewNodeCVELoader(nodeCVEDataStore.Singleton())
+	})
+}
+
+// NewNodeCVELoader creates a new loader for nodeCVE data.
+func NewNodeCVELoader(ds nodeCVEDataStore.DataStore) NodeCVELoader {
+	return &nodeCVELoaderImpl{
+		loaded: make(map[string]*storage.NodeCVE),
+		ds:     ds,
+	}
+}
+
+// GetNodeCVELoader returns the NodeCVELoader from the context if it exists.
+func GetNodeCVELoader(ctx context.Context) (NodeCVELoader, error) {
+	loader, err := GetLoader(ctx, nodeCveLoaderType)
+	if err != nil {
+		return nil, err
+	}
+	return loader.(NodeCVELoader), nil
+}
+
+// NodeCVELoader loads nodeCVE data, and stores already loaded nodeCVEs for other ops in the same context to use.
+type NodeCVELoader interface {
+	FromIDs(ctx context.Context, ids []string) ([]*storage.NodeCVE, error)
+	FromID(ctx context.Context, id string) (*storage.NodeCVE, error)
+	FromQuery(ctx context.Context, query *v1.Query) ([]*storage.NodeCVE, error)
+
+	CountFromQuery(ctx context.Context, query *v1.Query) (int32, error)
+	CountAll(ctx context.Context) (int32, error)
+}
+
+// nodeCVELoaderImpl implements the NodeCVELoader interface.
+type nodeCVELoaderImpl struct {
+	lock   sync.RWMutex
+	loaded map[string]*storage.NodeCVE
+
+	ds nodeCVEDataStore.DataStore
+}
+
+//func enrichNodeCVESeverity(value *storage.NodeCVE) {
+//	value.Severity = cvss.VulnToSeverity(cvss.NewFromNodeCVE(value))
+//}
+
+// FromIDs loads a set of nodeCVEs from a set of ids.
+func (idl *nodeCVELoaderImpl) FromIDs(ctx context.Context, ids []string) ([]*storage.NodeCVE, error) {
+	nodeCves, err := idl.load(ctx, ids)
+	if err != nil {
+		return nil, err
+	}
+	return nodeCves, nil
+}
+
+// FromID loads a nodeCVE from an ID.
+func (idl *nodeCVELoaderImpl) FromID(ctx context.Context, id string) (*storage.NodeCVE, error) {
+	nodeCves, err := idl.load(ctx, []string{id})
+	if err != nil {
+		return nil, err
+	}
+	return nodeCves[0], nil
+}
+
+// FromQuery loads a set of nodeCVEs that match a query.
+func (idl *nodeCVELoaderImpl) FromQuery(ctx context.Context, query *v1.Query) ([]*storage.NodeCVE, error) {
+	results, err := idl.ds.Search(ctx, query)
+	if err != nil {
+		return nil, err
+	}
+	return idl.FromIDs(ctx, search.ResultsToIDs(results))
+}
+
+func (idl *nodeCVELoaderImpl) CountFromQuery(ctx context.Context, query *v1.Query) (int32, error) {
+	count, err := idl.ds.Count(ctx, query)
+	if err != nil {
+		return 0, err
+	}
+	return int32(count), nil
+}
+
+func (idl *nodeCVELoaderImpl) CountAll(ctx context.Context) (int32, error) {
+	count, err := idl.ds.Count(ctx, search.EmptyQuery())
+	return int32(count), err
+}
+
+func (idl *nodeCVELoaderImpl) load(ctx context.Context, ids []string) ([]*storage.NodeCVE, error) {
+	nodeCves, missing := idl.readAll(ids)
+	if len(missing) > 0 {
+		var err error
+		nodeCves, err = idl.ds.GetBatch(ctx, collectMissing(ids, missing))
+		if err != nil {
+			return nil, err
+		}
+		for _, cve := range nodeCves {
```

Comment:
This is not required in Postgres because each instance of node vulns is unique to only one distro and therefore the stored top-level cvss and severity fields hold the final required values.